// anime-info/dist/onepunchman/utils/lang.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var tags_enum_1 = require("../enums/tags.enum");
function getTagByLang(lang) {
    switch (lang) {
        case 'en':
            return {
                abilities: tags_enum_1.TagsEn.Abilities,
                affiliation: tags_enum_1.TagsEn.Affiliation,
                age: tags_enum_1.TagsEn.Age,
                alias: tags_enum_1.TagsEn.Alias,
                class: tags_enum_1.TagsEn.Class,
                description: tags_enum_1.TagsEn.Description,
                gallery: tags_enum_1.TagsEn.Gallery,
                height: tags_enum_1.TagsEn.Height,
                location: tags_enum_1.TagsEn.Location,
                name: tags_enum_1.TagsEn.Name,
                occupation: tags_enum_1.TagsEn.Occupation,
                partner: tags_enum_1.TagsEn.Partner,
                photo: tags_enum_1.TagsEn.Photo,
                race: tags_enum_1.TagsEn.Race,
                rank: tags_enum_1.TagsEn.Rank,
                sex: tags_enum_1.TagsEn.Sex,
                status: tags_enum_1.TagsEn.Status,
                weight: tags_enum_1.TagsEn.Weight
            };
        case 'es':
            return {
                abilities: tags_enum_1.TagsEs.Abilities,
                affiliation: tags_enum_1.TagsEs.Affiliation,
                age: tags_enum_1.TagsEs.Age,
                alias: tags_enum_1.TagsEs.Alias,
                class: tags_enum_1.TagsEs.Class,
                description: tags_enum_1.TagsEs.Description,
                gallery: tags_enum_1.TagsEs.Gallery,
                height: tags_enum_1.TagsEs.Height,
                location: tags_enum_1.TagsEs.Location,
                name: tags_enum_1.TagsEs.Name,
                occupation: tags_enum_1.TagsEs.Occupation,
                partner: tags_enum_1.TagsEs.Partner,
                photo: tags_enum_1.TagsEs.Photo,
                race: tags_enum_1.TagsEs.Race,
                rank: tags_enum_1.TagsEs.Rank,
                sex: tags_enum_1.TagsEs.Sex,
                status: tags_enum_1.TagsEs.Status,
                weight: tags_enum_1.TagsEs.Weight
            };
        case 'pt-br':
        default:
            throw new Error('language not available');
    }
}
exports.getTagByLang = getTagByLang;
