// anime-info/dist/dragonball/enums/tags.enum.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var TagsPtBr;
(function (TagsPtBr) {
    TagsPtBr["Name"] = ".page-header__title";
    TagsPtBr["Description"] = "#mw-content-text > aside ~ p";
    TagsPtBr["Sex"] = "[data-source=\"Sexo\"] > div";
    TagsPtBr["Birthday"] = "[data-source=\"Nasceu\"] > div > a";
    TagsPtBr["Height"] = "[data-source=\"Altura\"] > div";
    TagsPtBr["Weight"] = "[data-source=\"Peso\"] > div";
    TagsPtBr["Alias"] = "[data-source=\"NomeAlternativo\"] > div";
    TagsPtBr["Race"] = "[data-source=\"Ra\u00E7a\"] > div > a";
    TagsPtBr["Death"] = "[data-source=\"Morreu\"] > div a";
    TagsPtBr["Address"] = "[data-source=\"Endere\u00E7o\"] > div > a";
    TagsPtBr["Occupation"] = "[data-source=\"Ocupa\u00E7\u00E3o\"] > div";
    TagsPtBr["Relatives"] = "[data-source=\"Fam\u00EDlia\"] > div > li";
    TagsPtBr["Photo"] = "[data-source='imagem'] > div figure a|[data-source='imagem'] a";
    TagsPtBr["Appears"] = "[data-source=\"Apareceu em\"] > div > div > a";
    TagsPtBr["Transformation"] = ".show-info-icon";
})(TagsPtBr = exports.TagsPtBr || (exports.TagsPtBr = {}));
var TagsEn;
(function (TagsEn) {
    TagsEn["Name"] = ".page-header__title";
    TagsEn["Description"] = "#mw-content-text > aside ~ p";
    TagsEn["Sex"] = "[data-source=\"Gender\"] > div ";
    TagsEn["Birthday"] = "[data-source=\"Date of birth\"] > div > a ";
    TagsEn["Height"] = "[data-source=\"Height\"] > div";
    TagsEn["Weight"] = "[data-source=\"Weight\"] > div";
    TagsEn["Alias"] = "[data-source=\"AltName\"] > div";
    TagsEn["Race"] = "[data-source=\"Race\"] > div > a";
    TagsEn["Death"] = "[data-source=\"Date of death\"] > div a ";
    TagsEn["Address"] = "[data-source=\"Address\"] > div > a";
    TagsEn["Occupation"] = "[data-source=\"Occupation\"] > div";
    TagsEn["Relatives"] = "[data-source=\"FamConnect\"] > div > a";
    TagsEn["Photo"] = "[data-source=\"image\"] > div figure a|[data-source=\"image\"] a";
    TagsEn["Appears"] = "[data-source=\"Appears in\"] div > div > div > a";
    TagsEn["Transformation"] = "[title=\"Forms and transformations\"] > figure";
})(TagsEn = exports.TagsEn || (exports.TagsEn = {}));
var TagsEs;
(function (TagsEs) {
    TagsEs["Name"] = ".page-header__title";
    TagsEs["Description"] = "#mw-content-text > aside ~ p";
    TagsEs["Sex"] = "[data-source=\"sexo\"] > div";
    TagsEs["Birthday"] = "[data-source=\"fecha de nacimiento\"] > div > a";
    TagsEs["Height"] = "[data-source=\"Altura\"] > div";
    TagsEs["Weight"] = "[data-source=\"Peso\"] > div";
    TagsEs["Alias"] = "[data-source=\"otros nombres\"] > div > li";
    TagsEs["Race"] = "[data-source=\"raza\"] > div > a";
    TagsEs["Death"] = "[data-source=\"fecha de muerte\"] > div a";
    TagsEs["Address"] = "[data-source=\"residencia\"] > div a";
    TagsEs["Relatives"] = "[data-source=\"ocupaci\u00F3n\"] > div > li";
    TagsEs["Family"] = "[data-source=\"familia\"] > div > li";
    TagsEs["Photo"] = ".pi-image-collection > div figure a|.pi-image a";
    TagsEs["Appears"] = "[data-source=\"saga\"] > div > a";
    TagsEs["Transformation"] = "[data-source=\"transformaciones\"] > div > li";
})(TagsEs = exports.TagsEs || (exports.TagsEs = {}));
