// anime-info/dist/onepunchman/enums/tags.enum.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var TagsPtBr;
(function (TagsPtBr) {
})(TagsPtBr = exports.TagsPtBr || (exports.TagsPtBr = {}));
var TagsEn;
(function (TagsEn) {
    TagsEn["Abilities"] = "[data-source=\"abilities\"] > div";
    TagsEn["Affiliation"] = "[data-source=\"affiliation\"] > div";
    TagsEn["Age"] = "[data-source='age'] > div";
    TagsEn["Alias"] = "[data-source=\"alias\"] > div";
    TagsEn["Class"] = "[data-source=\"class\"] > div";
    TagsEn["Description"] = "#mw-content-text > p";
    TagsEn["Gallery"] = "#mw-content-text img:not(.sprite)";
    TagsEn["Height"] = "[data-source=\"height\"] > div";
    TagsEn["Location"] = "[data-source=\"location\"] > div a";
    TagsEn["Name"] = ".page-header__title";
    TagsEn["Occupation"] = "[data-source=\"occupation\"] > div";
    TagsEn["Partner"] = "[data-source=\"partners\"] > div";
    TagsEn["Photo"] = ".pi-image-collection > div figure a|.pi-image a";
    TagsEn["Race"] = "[data-source=\"race\"] > div > a";
    TagsEn["Rank"] = "[data-source=\"rank\"] > div";
    TagsEn["Sex"] = "[data-source=\"gender\"] > div";
    TagsEn["Status"] = "[data-source=\"status\"] > div";
    TagsEn["Weight"] = "[data-source=\"weight\"] > div";
})(TagsEn = exports.TagsEn || (exports.TagsEn = {}));
var TagsEs;
(function (TagsEs) {
    TagsEs["Abilities"] = "[data-source=\"habilidades\"] > div";
    TagsEs["Affiliation"] = "[data-source=\"afiliacion\"] > div";
    TagsEs["Age"] = "[data-source='edad'] > div";
    TagsEs["Alias"] = "[data-source=\"alias\"] > div";
    TagsEs["Class"] = "[data-source=\"clase\"] > div";
    TagsEs["Description"] = "#mw-content-text > p";
    TagsEs["Gallery"] = "#mw-content-text img:not(.sprite)";
    TagsEs["Height"] = "[data-source='altura'] > div";
    TagsEs["Location"] = "[data-source=\"ciudad\"] > div a";
    TagsEs["Name"] = ".page-header__title";
    TagsEs["Occupation"] = "[data-source=\"ocupacion\"] > div";
    TagsEs["Partner"] = "[data-source=\"compa\u00F1ero\"] > div";
    TagsEs["Photo"] = ".pi-image-collection > div figure a|.pi-image a";
    TagsEs["Race"] = "[data-source=\"raza\"] > div";
    TagsEs["Rank"] = "[data-source=\"rango\"] > div";
    TagsEs["Sex"] = "[data-source=\"genero\"] > div";
    TagsEs["Status"] = "[data-source=\"estado\"] > div";
    TagsEs["Weight"] = "[data-source='peso'] > div";
})(TagsEs = exports.TagsEs || (exports.TagsEs = {}));
