// anime-info/dist/hunterxhunter/utils/lang.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var tags_enum_1 = require("../enums/tags.enum");
function getTagByLang(lang) {
    switch (lang) {
        case 'en':
            return {
                abilities: tags_enum_1.TagsEn.Abilities,
                abilities_description: tags_enum_1.TagsEn.AbilitiesDescription,
                affiliation: tags_enum_1.TagsEn.Affiliation,
                age: tags_enum_1.TagsEn.Age,
                alias: tags_enum_1.TagsEn.Alias,
                birthday: tags_enum_1.TagsEn.Birthday,
                blood_type: tags_enum_1.TagsEn.BloodType,
                description: tags_enum_1.TagsEn.Description,
                eye_color: tags_enum_1.TagsEn.EyeColor,
                gender: tags_enum_1.TagsEn.Gender,
                hair_color: tags_enum_1.TagsEn.HairColor,
                height: tags_enum_1.TagsEn.Height,
                japanese_voice: tags_enum_1.TagsEn.JapaneseVoice,
                // class: TagsEn.Class,
                // gallery: TagsEn.Gallery,
                // location: TagsEn.Location,
                name: tags_enum_1.TagsEn.Name,
                nen_type: tags_enum_1.TagsEn.NenType,
                occupation: tags_enum_1.TagsEn.Occupation,
                relatives: tags_enum_1.TagsEn.Relatives,
                photo: tags_enum_1.TagsEn.Photo,
                // partner: TagsEn.Partner,
                // photo: TagsEn.Photo,
                // race: TagsEn.Race,
                // rank: TagsEn.Rank,
                // sex: TagsEn.Sex,
                status: tags_enum_1.TagsEn.Status,
                weight: tags_enum_1.TagsEn.Weight
            };
        case 'es':
            return {
                abilities: tags_enum_1.TagsEs.Abilities,
                abilities_description: tags_enum_1.TagsEs.AbilitiesDescription,
                affiliation: tags_enum_1.TagsEs.Affiliation,
                age: tags_enum_1.TagsEs.Age,
                alias: tags_enum_1.TagsEs.Alias,
                birthday: tags_enum_1.TagsEs.Birthday,
                blood_type: tags_enum_1.TagsEs.BloodType,
                description: tags_enum_1.TagsEs.Description,
                eye_color: tags_enum_1.TagsEs.EyeColor,
                gender: tags_enum_1.TagsEs.Gender,
                hair_color: tags_enum_1.TagsEs.HairColor,
                japanese_voice: tags_enum_1.TagsEs.JapaneseVoice,
                // class: TagsEs.Class,
                // gallery: TagsEs.Gallery,
                height: tags_enum_1.TagsEs.Height,
                // location: TagsEs.Location,
                name: tags_enum_1.TagsEs.Name,
                nen_type: tags_enum_1.TagsEs.NenType,
                occupation: tags_enum_1.TagsEs.Occupation,
                relatives: tags_enum_1.TagsEs.Relatives,
                photo: tags_enum_1.TagsEs.Photo,
                // partner: TagsEs.Partner,
                // photo: TagsEs.Photo,
                // race: TagsEs.Race,
                // rank: TagsEs.Rank,
                // sex: TagsEs.Sex,
                status: tags_enum_1.TagsEs.Status,
                weight: tags_enum_1.TagsEs.Weight
            };
        case 'pt-br':
        default:
            throw new Error('language not available');
    }
}
exports.getTagByLang = getTagByLang;
function getBaseUrl(lang) {
    var langs = {
        'pt-br': 'https://hunter-x-hunter.fandom.com/pt/wiki/',
        es: 'https://hunterxhunter.fandom.com/es/wiki/',
        en: 'https://hunterxhunter.fandom.com/wiki/'
    };
    return langs[lang];
}
exports.getBaseUrl = getBaseUrl;
