// anime-info/dist/hunterxhunter/enums/tags.enum.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var TagsPtBr;
(function (TagsPtBr) {
})(TagsPtBr = exports.TagsPtBr || (exports.TagsPtBr = {}));
var TagsEn;
(function (TagsEn) {
    TagsEn["Abilities"] = "[data-source=\"abilities\"] > div";
    TagsEn["AbilitiesDescription"] = "table[class=\"wikitable\"][cellpadding=\"2\"][cellspacing=\"1/6\"] > tbody > tr";
    TagsEn["Affiliation"] = "[data-source=\"affiliation\"] > div|[data-source=\"previous affiliation\"] > div";
    TagsEn["Age"] = "[data-source=\"age\"] > div";
    TagsEn["Alias"] = "[data-source=\"kana\"] > div|[data-source=\"r\u014Dmaji\"] > div";
    TagsEn["Birthday"] = "[data-source=\"birthday\"] > div > a > span";
    TagsEn["BloodType"] = "[data-source=\"blood\"] > div";
    TagsEn["Description"] = "#mw-content-text > aside ~ p";
    TagsEn["EyeColor"] = "[data-source=\"eyes\"] > div";
    TagsEn["Gender"] = "[data-source=\"gender\"] > div";
    TagsEn["HairColor"] = "[data-source=\"hair\"] > div";
    TagsEn["Height"] = "[data-source=\"height\"] > div";
    TagsEn["JapaneseVoice"] = "[data-source=\"japanese voice\"] > div";
    TagsEn["Name"] = ".page-header__title";
    TagsEn["NenType"] = "[data-source=\"type\"] > div";
    TagsEn["Occupation"] = "[data-source=\"occupation\"] > div|[data-source=\"previous occupation\"] > div";
    TagsEn["Photo"] = "[data-source=\"image\"] > div figure a|[data-source=\"image\"] a";
    TagsEn["Relatives"] = "[data-source=\"relatives\"] > div";
    TagsEn["Status"] = "[data-source=\"status\"] > div";
    TagsEn["Weight"] = "[data-source=\"weight\"] > div";
})(TagsEn = exports.TagsEn || (exports.TagsEn = {}));
var TagsEs;
(function (TagsEs) {
    TagsEs["Abilities"] = "[data-source=\"habilidades\"] > div";
    TagsEs["AbilitiesDescription"] = "table[class=\"wikitable\"][cellpadding=\"2\"][cellspacing=\"1/6\"] > tbody > tr";
    TagsEs["Affiliation"] = "[data-source=\"afiliaci\u00F3n\"] > div";
    TagsEs["Age"] = "[data-source=\"edad\"] > div";
    TagsEs["Alias"] = "[data-source=\"kana\"] > div|[data-source=\"romaji\"] > div";
    TagsEs["Birthday"] = "[data-source=\"cumplea\u00F1os\"] > div";
    TagsEs["BloodType"] = "[data-source=\"tipo de sangre\"] > div";
    TagsEs["Description"] = "#mw-content-text > aside ~ p";
    TagsEs["EyeColor"] = "[data-source=\"color de ojos\"] > div";
    TagsEs["Gender"] = "[data-source=\"genero\"] > div";
    TagsEs["HairColor"] = "[data-source=\"color de pelo\"] > div";
    TagsEs["Height"] = "[data-source=\"altura\"] > div";
    TagsEs["JapaneseVoice"] = "[data-source=\"japones\"] > div";
    TagsEs["Name"] = ".page-header__title";
    TagsEs["NenType"] = "[data-source=\"tipo\"] > div";
    TagsEs["Occupation"] = "[data-source=\"ocupacion\"] > div";
    TagsEs["Photo"] = "[data-source=\"image\"] > div figure a|[data-source=\"image\"] a";
    TagsEs["Relatives"] = "[data-source=\"familia\"] > div";
    TagsEs["Status"] = "[data-source=\"status\"] > div";
    TagsEs["Weight"] = "[data-source=\"peso\"] > div";
})(TagsEs = exports.TagsEs || (exports.TagsEs = {}));
