// anime-info/dist/naruto/utils/dom.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __generator = (this && this.__generator) || function (thisArg, body) {
    var _ = { label: 0, sent: function() { if (t[0] & 1) throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
    return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function() { return this; }), g;
    function verb(n) { return function (v) { return step([n, v]); }; }
    function step(op) {
        if (f) throw new TypeError("Generator is already executing.");
        while (_) try {
            if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done) return t;
            if (y = 0, t) op = [op[0] & 2, t.value];
            switch (op[0]) {
                case 0: case 1: t = op; break;
                case 4: _.label++; return { value: op[1], done: false };
                case 5: _.label++; y = op[1]; op = [0]; continue;
                case 7: op = _.ops.pop(); _.trys.pop(); continue;
                default:
                    if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) { _ = 0; continue; }
                    if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) { _.label = op[1]; break; }
                    if (op[0] === 6 && _.label < t[1]) { _.label = t[1]; t = op; break; }
                    if (t && _.label < t[2]) { _.label = t[2]; _.ops.push(op); break; }
                    if (t[2]) _.ops.pop();
                    _.trys.pop(); continue;
            }
            op = body.call(thisArg, _);
        } catch (e) { op = [6, e]; y = 0; } finally { f = t = 0; }
        if (op[0] & 5) throw op[1]; return { value: op[0] ? op[1] : void 0, done: true };
    }
};
Object.defineProperty(exports, "__esModule", { value: true });
var elements_dom_1 = require("../../shared/utils/elements.dom");
function getNameAndPhotoTable(document, tag, lang, url) {
    return __awaiter(this, void 0, void 0, function () {
        var dom, elements, list, index, icon;
        return __generator(this, function (_a) {
            switch (_a.label) {
                case 0:
                    if (!(lang === 'pt-br')) return [3 /*break*/, 2];
                    return [4 /*yield*/, elements_dom_1.getDom(url)];
                case 1:
                    dom = _a.sent();
                    elements = dom.querySelectorAll(tag);
                    list = [];
                    for (index = 1; index < elements.length; index += 1) {
                        icon = 'Sem Imagem';
                        if (elements[index].children[1] &&
                            elements[index].children[1].children[0] &&
                            elements[index].children[1].children[0].children[0]) {
                            icon = elements[index].children[1].children[0].children[0].getAttribute('data-src');
                            if (icon == null) {
                                icon = elements[index].children[1].children[0].children[0].getAttribute('src');
                            }
                        }
                        list.push({
                            name: elements_dom_1.removeSpacesString(elements[index].children[0].textContent),
                            icon: icon
                        });
                    }
                    return [2 /*return*/, list];
                case 2: return [2 /*return*/, elements_dom_1.getListElement(document, tag)];
            }
        });
    });
}
exports.getNameAndPhotoTable = getNameAndPhotoTable;
function getVoices(document, country, name) {
    var voices = [];
    var tagCountry = country.split('|');
    var tagName = name.split('|');
    tagCountry.forEach(function (value, index) {
        voices.push({
            name: elements_dom_1.$Query(document, tagName[index]),
            country: elements_dom_1.$Query(document, tagCountry[index])
        });
    });
    return voices;
}
exports.getVoices = getVoices;
function getQuotes(tag, lang, url) {
    return __awaiter(this, void 0, void 0, function () {
        var list, dom, elements;
        return __generator(this, function (_a) {
            switch (_a.label) {
                case 0:
                    list = [];
                    if (!(lang === 'pt-br')) return [3 /*break*/, 2];
                    return [4 /*yield*/, elements_dom_1.getDom(url)];
                case 1:
                    dom = _a.sent();
                    elements = dom.querySelectorAll(tag);
                    elements.forEach(function (el) {
                        list.push({
                            phrase: elements_dom_1.getTextContent(el.children[0]),
                            recipient: elements_dom_1.getTextContent(el.children[1]),
                            situation: elements_dom_1.getTextContent(el.children[2])
                        });
                    });
                    list.shift();
                    _a.label = 2;
                case 2: return [2 /*return*/, list];
            }
        });
    });
}
exports.getQuotes = getQuotes;
