// anime-info/dist/naruto/utils/lang.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var tags_enum_1 = require("../enums/tags.enum");
function getTagByLang(lang) {
    switch (lang) {
        case 'pt-br':
            return {
                name: tags_enum_1.TagsPtBr.Name,
                description: tags_enum_1.TagsPtBr.Description,
                photo: tags_enum_1.TagsPtBr.Photo,
                titles: tags_enum_1.TagsPtBr.Titles,
                birthday: tags_enum_1.TagsPtBr.Birthday,
                status: tags_enum_1.TagsPtBr.Status,
                age: tags_enum_1.TagsPtBr.Age,
                height: tags_enum_1.TagsPtBr.Height,
                weight: tags_enum_1.TagsPtBr.Weight,
                classification: tags_enum_1.TagsPtBr.Classification,
                team: tags_enum_1.TagsPtBr.Team,
                partner: tags_enum_1.TagsPtBr.Partner,
                occupation: tags_enum_1.TagsPtBr.Occupation,
                ninja_rank: tags_enum_1.TagsPtBr.NinjaRank,
                ninja_registration: tags_enum_1.TagsPtBr.NinjaRegistration,
                family: tags_enum_1.TagsPtBr.Family,
                jutsu: tags_enum_1.TagsPtBr.Jutsu,
                nature_type: tags_enum_1.TagsPtBr.NatureType,
                tools: tags_enum_1.TagsPtBr.Tools,
                quotes: tags_enum_1.TagsPtBr.Quotes,
                sex: tags_enum_1.TagsPtBr.Sex,
                kekkei_genkai: tags_enum_1.TagsPtBr.KekkeiGenkai,
                affiliation: tags_enum_1.TagsPtBr.Affiliation,
                clan: tags_enum_1.TagsPtBr.Clan,
                anime: tags_enum_1.TagsPtBr.Anime,
                game: tags_enum_1.TagsPtBr.Game,
                manga: tags_enum_1.TagsPtBr.Manga,
                movie: tags_enum_1.TagsPtBr.Movie,
                novel: tags_enum_1.TagsPtBr.Novel,
                ova: tags_enum_1.TagsPtBr.Ova,
                voices: {
                    name: tags_enum_1.TagsPtBr.VoicesName,
                    country: tags_enum_1.TagsPtBr.VoicesCountry
                }
            };
        case 'es':
            return {
                name: tags_enum_1.TagsEs.Name,
                description: tags_enum_1.TagsEs.Description,
                photo: tags_enum_1.TagsEs.Photo,
                titles: tags_enum_1.TagsEs.Titles,
                birthday: tags_enum_1.TagsEs.Birthday,
                status: tags_enum_1.TagsEs.Status,
                age: tags_enum_1.TagsEs.Age,
                sex: tags_enum_1.TagsEs.Sex,
                height: tags_enum_1.TagsEs.Height,
                weight: tags_enum_1.TagsEs.Weight,
                classification: tags_enum_1.TagsEs.Classification,
                team: tags_enum_1.TagsEs.Team,
                partner: tags_enum_1.TagsEs.Partner,
                occupation: tags_enum_1.TagsEs.Occupation,
                ninja_rank: tags_enum_1.TagsEs.NinjaRank,
                ninja_registration: tags_enum_1.TagsEs.NinjaRegistration,
                family: tags_enum_1.TagsEs.Family,
                jutsu: tags_enum_1.TagsEs.Jutsu,
                nature_type: tags_enum_1.TagsEs.NatureType,
                tools: tags_enum_1.TagsEs.Tools,
                quotes: tags_enum_1.TagsEs.Quotes,
                kekkei_genkai: tags_enum_1.TagsEs.KekkeiGenkai,
                affiliation: tags_enum_1.TagsEs.Affiliation,
                clan: tags_enum_1.TagsEs.Clan,
                anime: tags_enum_1.TagsEs.Anime,
                game: tags_enum_1.TagsEs.Game,
                manga: tags_enum_1.TagsEs.Manga,
                movie: tags_enum_1.TagsEs.Movie,
                novel: tags_enum_1.TagsEs.Novel,
                ova: tags_enum_1.TagsEs.Ova,
                voices: {
                    name: tags_enum_1.TagsEs.VoicesName,
                    country: tags_enum_1.TagsEs.VoicesCountry
                }
            };
        case 'en':
        default:
            throw new Error('language not available');
    }
}
exports.getTagByLang = getTagByLang;
