// anime-info/dist/dragonball/utils/lang.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var tags_enum_1 = require("../enums/tags.enum");
function getTagByLang(lang) {
    switch (lang) {
        case 'pt-br':
            return {
                name: tags_enum_1.TagsPtBr.Name,
                description: tags_enum_1.TagsPtBr.Description,
                sex: tags_enum_1.TagsPtBr.Sex,
                birthday: tags_enum_1.TagsPtBr.Birthday,
                height: tags_enum_1.TagsPtBr.Height,
                weight: tags_enum_1.TagsPtBr.Weight,
                alias: tags_enum_1.TagsPtBr.Alias,
                race: tags_enum_1.TagsPtBr.Race,
                death: tags_enum_1.TagsPtBr.Death,
                address: tags_enum_1.TagsPtBr.Address,
                occupation: tags_enum_1.TagsPtBr.Occupation,
                relatives: tags_enum_1.TagsPtBr.Relatives,
                photo: tags_enum_1.TagsPtBr.Photo,
                appears: tags_enum_1.TagsPtBr.Appears,
                transformation: tags_enum_1.TagsPtBr.Transformation
            };
        case 'es':
            return {
                name: tags_enum_1.TagsEs.Name,
                description: tags_enum_1.TagsEs.Description,
                sex: tags_enum_1.TagsEs.Sex,
                birthday: tags_enum_1.TagsEs.Birthday,
                height: tags_enum_1.TagsEs.Height,
                weight: tags_enum_1.TagsEs.Weight,
                alias: tags_enum_1.TagsEs.Alias,
                race: tags_enum_1.TagsEs.Race,
                death: tags_enum_1.TagsEs.Death,
                address: tags_enum_1.TagsEs.Address,
                occupation: tags_enum_1.TagsEs.Relatives,
                relatives: tags_enum_1.TagsEs.Family,
                photo: tags_enum_1.TagsEs.Photo,
                appears: tags_enum_1.TagsEs.Appears,
                transformation: tags_enum_1.TagsEs.Transformation
            };
        case 'en':
            return {
                name: tags_enum_1.TagsEn.Name,
                description: tags_enum_1.TagsEn.Description,
                sex: tags_enum_1.TagsEn.Sex,
                birthday: tags_enum_1.TagsEn.Birthday,
                height: tags_enum_1.TagsEn.Height,
                weight: tags_enum_1.TagsEn.Weight,
                alias: tags_enum_1.TagsEn.Alias,
                race: tags_enum_1.TagsEn.Race,
                death: tags_enum_1.TagsEn.Death,
                address: tags_enum_1.TagsEn.Address,
                occupation: tags_enum_1.TagsEn.Occupation,
                relatives: tags_enum_1.TagsEn.Relatives,
                photo: tags_enum_1.TagsEn.Photo,
                appears: tags_enum_1.TagsEn.Appears,
                transformation: tags_enum_1.TagsEn.Transformation
            };
        default:
            throw new Error('language not available');
    }
}
exports.getTagByLang = getTagByLang;
