// anime-info/dist/naruto/enums/tags.enum.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var TagsPtBr;
(function (TagsPtBr) {
    TagsPtBr["Name"] = ".page-header__title";
    TagsPtBr["Description"] = "#mw-content-text > aside ~ p";
    TagsPtBr["Photo"] = "[data-source='Imagem'] > div figure a|[data-source='Imagem'] a";
    TagsPtBr["Titles"] = "[data-source='Nome'] > div > i";
    TagsPtBr["Age"] = "[data-source='IdadeI'] > div";
    TagsPtBr["Height"] = "[data-source='AlturaI'] > div";
    TagsPtBr["Weight"] = "[data-source='PesoI'] > div";
    TagsPtBr["Classification"] = "[data-source='Classifica\u00E7\u00E3o'] > div > a";
    TagsPtBr["Team"] = "[data-source='Times'] > div > a";
    TagsPtBr["Partner"] = "[data-source='Parceiro'] > div > a";
    TagsPtBr["Occupation"] = "[data-source='Ocupa\u00E7\u00E3o'] > div > a";
    TagsPtBr["NinjaRank"] = "[data-source='Patente NinjaI'] > div > a";
    TagsPtBr["NinjaRegistration"] = "[data-source='Registro Ninja'] > div";
    TagsPtBr["Family"] = "[data-source='Fam\u00EDlia'] > div > div > .NavContent > p > a";
    TagsPtBr["Jutsu"] = ".tabbertab[title='Mang\u00E1'] > div > table > tbody > tr";
    TagsPtBr["NatureType"] = "[data-source='Chakra'] > div > div > .NavContent > p > a";
    TagsPtBr["Tools"] = ".lua > tbody > tr";
    TagsPtBr["Quotes"] = ".lua > tbody > tr";
    TagsPtBr["Birthday"] = "[data-source=Anivers\u00E1rio] > div";
    TagsPtBr["Status"] = "[data-source=Estado] > div";
    TagsPtBr["Sex"] = "[data-source=Sexo] > div > a:nth-child(2)";
    TagsPtBr["Manga"] = "[data-source=Mang\u00E1] > div > a";
    TagsPtBr["Anime"] = "[data-source=Anime] > div";
    TagsPtBr["Novel"] = "[data-source=Novela] > div";
    TagsPtBr["Movie"] = "[data-source=Filme] > div > i > a";
    TagsPtBr["Game"] = "[data-source=Game] > div > i > a";
    TagsPtBr["Ova"] = "[data-source=Ova] > div > i > a";
    TagsPtBr["Clan"] = "[data-source='Cl\u00E3'] > div > a";
    TagsPtBr["Affiliation"] = "[data-source='Afilia\u00E7\u00E3o'] > div > a";
    TagsPtBr["KekkeiGenkai"] = "[data-source='Kekkei Genkai'] > div > a";
    TagsPtBr["VoicesName"] = "[data-source=Japon\u00EAs] > div|[data-source=Portugu\u00EAs] > div";
    TagsPtBr["VoicesCountry"] = "[data-source=Japon\u00EAs] > h3 > small > b|[data-source=Portugu\u00EAs] > h3 > small > b";
})(TagsPtBr = exports.TagsPtBr || (exports.TagsPtBr = {}));
var TagsEn;
(function (TagsEn) {
    TagsEn["Name"] = "";
    TagsEn["Description"] = "";
    TagsEn["Photo"] = "";
    TagsEn["Titles"] = "";
    TagsEn["Age"] = "";
    TagsEn["Height"] = "";
    TagsEn["Weight"] = "";
    TagsEn["Classification"] = "";
    TagsEn["Team"] = "";
    TagsEn["Partner"] = "";
    TagsEn["Occupation"] = "";
    TagsEn["NinjaRank"] = "";
    TagsEn["NinjaRegistration"] = "";
    TagsEn["Family"] = "";
    TagsEn["Jutsu"] = "";
    TagsEn["NatureType"] = "";
    TagsEn["Tools"] = "";
    TagsEn["Quotes"] = "";
    TagsEn["Birthday"] = "";
    TagsEn["Status"] = "";
    TagsEn["Sex"] = "";
    TagsEn["Manga"] = "";
    TagsEn["Anime"] = "";
    TagsEn["Novel"] = "";
    TagsEn["Movie"] = "";
    TagsEn["Game"] = "";
    TagsEn["Ova"] = "";
    TagsEn["Clan"] = "";
    TagsEn["Affiliation"] = "";
    TagsEn["KekkeiGenkai"] = "";
    TagsEn["VoicesName"] = "";
    TagsEn["VoicesCountry"] = "";
})(TagsEn = exports.TagsEn || (exports.TagsEn = {}));
var TagsEs;
(function (TagsEs) {
    TagsEs["Name"] = ".page-header__title";
    TagsEs["Description"] = "#mw-content-text > aside ~ p";
    TagsEs["Photo"] = ".pi-image-collection > div figure a|.pi-image a";
    TagsEs["Titles"] = "[data-source='Sobrenombre'] > div > i";
    TagsEs["Age"] = "[data-source='Edad'] > div";
    TagsEs["Height"] = "[data-source='Altura'] > div";
    TagsEs["Weight"] = "[data-source='Peso'] > div";
    TagsEs["Classification"] = "[data-source='Clasificacion'] > div > a";
    TagsEs["Team"] = "[data-source='Equipo'] > div > a";
    TagsEs["Partner"] = "[data-source='Compa\u00F1ero'] > div > a";
    TagsEs["Occupation"] = "[data-source='Ocupacion'] div > a";
    TagsEs["NinjaRank"] = "[data-source='Rango'] div > a";
    TagsEs["NinjaRegistration"] = "[data-source='Registro'] > div";
    TagsEs["Family"] = "[data-source='Fam\u00EDlia'] > div > a";
    TagsEs["Jutsu"] = "[data-source='Jutsus'] > div > a";
    TagsEs["NatureType"] = "[data-source='Elemento'] > div > a";
    TagsEs["Tools"] = "[data-source='Armas'] > div > a";
    TagsEs["Quotes"] = "#mw-content-text > ul";
    TagsEs["Birthday"] = "[data-source=Nacimiento] > div";
    TagsEs["Status"] = "[data-source=Nacimiento] ~ div:nth-of-type(3) > div";
    TagsEs["Sex"] = "[data-source=Nacimiento] ~ div:nth-of-type(2) > div > a:nth-child(2)";
    TagsEs["Manga"] = "[data-source=Debutmanga] > div > a";
    TagsEs["Anime"] = "[data-source=Debutanime] > div > a";
    TagsEs["Novel"] = "[data-source=Debutnovela] > div > a";
    TagsEs["Movie"] = "[data-source=Debutpelicula] > div > a";
    TagsEs["Game"] = "[data-source=Debutjuego] > div > a";
    TagsEs["Ova"] = "[data-source=Debutova] > div > a";
    TagsEs["Clan"] = "[data-source='Clan'] > div > a";
    TagsEs["Affiliation"] = "[data-source='Afiliacion'] > div > a";
    TagsEs["KekkeiGenkai"] = "[data-source='Kekkeigenkai'] > div > a";
    TagsEs["VoicesName"] = "[data-source=Seiy\u016B] > div|[data-source=Espa\u00F1ollatino] > div|[data-source=Espa\u00F1olespa\u00F1a] > div";
    TagsEs["VoicesCountry"] = "[data-source=Seiy\u016B] > h3|[data-source=Espa\u00F1ollatino] > h3|[data-source=Espa\u00F1olespa\u00F1a] > h3";
})(TagsEs = exports.TagsEs || (exports.TagsEs = {}));
